src/main/java/com/minecolonies/coremod/colony/crafting/CustomRecipeManager.java

```diff
@@ -79,13 +115,21 @@ public void removeRecipe(@NotNull final ResourceLocation toRemove)
     }
 
     /**
-     * The complete list of custom recipes, by crafter. 
+     * The complete list of custom recipes, by crafter.
      */
     public Map<String, Map<ResourceLocation, CustomRecipe>> getAllRecipes()
     {
         return recipeMap;
     }
 
+    /**
+     * Get the custom recipes for an item, or null if no matching recipe exists.
+     */
+    public List<CustomRecipe> getRecipeByOutput(final Item item)
```

Comment:
This isn't enough, this should support an ItemStorage as the search criteria, so that it can find the recipe with the itemstack that matches.